app/models/event.rb

```diff
@@ -0,0 +1,12 @@
+class Event < ApplicationRecord
+  validates :location, presence: true
+  validates :date, presence: true
+  validates :description, presence: true
+
+  scope :past, -> { where('date < ?', Date.current) }
+  scope :upcoming, -> { where('date >= ?', Date.current) }
+
+  belongs_to :creator, foreign_key: :user_id, class_name: 'User'
```

Comment:
- [ ] Fix the foreign key to be `creator_id` and remember to create a migration to change the schema to the correct foreign key name.